staging/src/k8s.io/kubectl/pkg/cmd/debug/debug.go

```diff
@@ -272,8 +272,16 @@ func (o *DebugOptions) Validate(cmd *cobra.Command) error {
 	}
 
 	// TargetContainer
-	if len(o.TargetContainer) > 0 && len(o.CopyTo) > 0 {
-		return fmt.Errorf("--target is incompatible with --copy-to. Use --share-processes instead.")
+	if len(o.TargetContainer) > 0 {
+		if len(o.CopyTo) > 0 {
+			return fmt.Errorf("--target is incompatible with --copy-to. Use --share-processes instead.")
+		}
+		if !o.Quiet {
+			// If the runtime doesn't support container namespace targeting this will fail silently, which has caused
+			// some confusion (ex: https://issues.k8s.io/98362), so print a warning. This can be removed when
+			// EphemeralContainers are generally available.
+			fmt.Fprintf(o.Out, "WARNING: Targeting container %q. Some container runtimes do not support this feature.\n", o.TargetContainer)
```

Comment:
Is there anything more specific we can add here?

Something like "If you don't see the processes you're expecting know that some container runtimes do not support this feature".

Or a link to which ones are supported?

Reply:
Definitely can reword as you can suggested. That makes sense, and matches the style for `kubectl attach` (iirc).

We don't have a list of which runtimes support this feature, unfortunately,  but that would be nice. I wonder if there's a conformance test suite I can use to figure that out...

Reply:
@eddiezane Updated, PTAL.